src/MarginTrading.MarketMaker/Controllers/ExtPriceExchangesController.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using AutoMapper;
+using JetBrains.Annotations;
+using MarginTrading.MarketMaker.Contracts;
+using MarginTrading.MarketMaker.Contracts.Models;
+using MarginTrading.MarketMaker.Infrastructure;
+using MarginTrading.MarketMaker.Models.Settings;
+using MarginTrading.MarketMaker.Services.ExtPrices;
+using Microsoft.AspNetCore.Mvc;
+
+namespace MarginTrading.MarketMaker.Controllers
+{
+    [Route("api/[controller]")]
+    public class ExtPriceExchangesController : Controller
+    {
+        private readonly IExtPricesSettingsService _extPricesSettingsService;
+        private readonly IConvertService _convertService;
+
+        public ExtPriceExchangesController(IExtPricesSettingsService extPricesSettingsService, IConvertService convertService)
+        {
+            _extPricesSettingsService = extPricesSettingsService;
+            _convertService = convertService;
+        }
+
+        /// <summary>
+        /// Gets all exchanges
+        /// </summary>
+        [HttpGet]
+        public IReadOnlyList<ExchangeExtPriceSettingsModel> List()
+        {
+            return _extPricesSettingsService.GetExchanges()
+                .SelectMany(a => a.Value.Select(e => Convert(a.Key, e.Key, e.Value)))
+                .OrderBy(a => a.AssetPairId).ThenBy(a => a.ExchangeName)
+                .ToList();
+        }
```

Comment:
Don't you think it is better to return object with the same structure that the stored one (dictionary)? In such case we will not need custom convert and it will be more user friendly result?